File: 231127/연속부분수열일까/contiguous-array-or-not.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws NumberFormatException, IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n1 = Integer.parseInt(st.nextToken());
        int n2 = Integer.parseInt(st.nextToken());
        int[] a = new int[n1];
        int[] b = new int[n2];
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n1; i++) {
            a[i] = Integer.parseInt(st.nextToken());
        }
        st = new StringTokenizer(br.readLine());
        for (int i = 0 ; i < n2; i++) {
            b[i] = Integer.parseInt(st.nextToken());
        }
        
        String answer = "No";
        for (int aIdx = 0; aIdx < n1; aIdx++) {
            int count = 0;
            for (int bIdx = 0; bIdx < n2; bIdx++) {
                if (aIdx + bIdx >= n1) break;
                if (a[aIdx + bIdx] != b[bIdx]) break;
                count++;
            }
            if (count == n2) {
                answer = "Yes";
                break;
            }
        }
        System.out.println(answer);
    }
}

File: 231127/연속부분수열일까/contiguous-array-or-not.js
const fs = require('fs');
const input = 
    fs.readFileSync(0).toString().trim().split('\n');
const [n1, n2] = input[0].split(' ').map(ele => Number.parseInt(ele));
const A = input[1].split(' ').map(ele => Number.parseInt(ele));
const B = input[2].split(' ').map(ele => Number.parseInt(ele));

let answer = 'No';
for (let AIdx = 0; AIdx < A.length; AIdx++) {
    let index = AIdx;
    let count = 0;
    for (let BIdx = 0; BIdx < B.length; BIdx++) {
        if (A[index] !== B[BIdx]) break;
        index++;
        count++;
    }
    if (count === B.length) {
        answer = 'Yes';
        break;
    }
}

console.log(answer);